sources/en/a/allnovel.py

```diff
@@ -59,77 +60,30 @@ def read_novel_info(self):
-        # page_count = 0
-        # pagination_link = soup.select_one('#list-chapter .pagination .last a')
-        # if isinstance(pagination_link, Tag):
-        #     page_count = int(str(pagination_link['data-page']))
-        # logger.info('Chapter list pages: %d' % page_count)
-
-        # futures = []
-        # for page in range(1, page_count + 2):
-        #     if page == 1:
-        #         f = self.executor.submit(lambda: soup)
-        #     else:
-        #         url = self.novel_url.split('?')[0].strip('/')
-        #         url += '?page=%d' % page
-        #         f = self.executor.submit(self.get_soup, url)
-        #     # end if
-        #     futures.append(f)
-        # # end for
-
-        # for i, f in enumerate(futures):
-        #     try:
-        #         soup = f.result()
-        #     except KeyboardInterrupt:
-        #         c = len([f.cancel() for f in futures[i:]])
-        #         logger.info('Cancelled remaining %d jobs', c)
-        #     # end try
-        #     for a in soup.select('ul.list-chapter li a'):
-        #         chap_id = len(self.chapters) + 1
-        #         vol_id = 1 + len(self.chapters) // 100
-        #         if len(self.volumes) < vol_id:
-        #             self.volumes.append({ 'id': vol_id })
-        #         # end if
-        #         self.chapters.append({
-        #             'id': chap_id,
-        #             'volume': vol_id,
-        #             'title': a['title'],
-        #             'url': self.absolute_url(a['href']),
-        #         })
-        #     # end for
-        # # end for
-
         possible_id = soup.select_one("input#truyen-id")
         assert possible_id, "No novel id"
+
         self.novel_id = possible_id["value"]
+
         soup = self.get_soup(
             self.home_url + "ajax-chapter-option?novelId=%s" % self.novel_id
         )
+
         for opt in soup.select("select option"):
             chap_id = len(self.chapters) + 1
-            vol_id = 1 + len(self.chapters) // 100
-            if len(self.volumes) < vol_id:
-                self.volumes.append({"id": vol_id})
             self.chapters.append(
                 {
                     "id": chap_id,
-                    "volume": vol_id,
                     "title": opt.text,
                     "url": self.absolute_url(opt["value"]),
                 }
             )
 
     def download_chapter_body(self, chapter):
         soup = self.get_soup(chapter["url"])
-        content = soup.select_one("div#chapter-content")
+        contents = soup.select_one("div#chapter-content")
 
-        assert isinstance(content, Tag), "No chapter content"
-        for child in content.contents:
-            child.extract()
-            if isinstance(child, Tag) and child.name == "h3":
-                break
+        assert isinstance(contents, Tag), "No chapter content"
+        self.cleaner.extract_contents(contents)
```

Comment:
use `clean_contents` if you are not returning in future changes

Reply:
Then it is necessary to change other crawlers too.

Reply:
def clean_contents(self, div) -> None: ??